warehouse/transformer/set.go

```diff
@@ -204,6 +205,24 @@ func storeRudderEvent(
 	if err != nil {
 		return fmt.Errorf("marshalling event: %w", err)
 	}
+	if t.config.populateSrcDestInfoInContext.Load() {
+		eventJSON, err = sjson.SetBytes(eventJSON, "context.sourceId", tec.event.Metadata.SourceID)
```

Comment:
Looks like this same logic is also present in `eventContext` method? Any reason why we need it at both the places?

Reply:
We need the raw events with those additional fields. Since we are not modifying the actual event, we need to set it here.